config/default.yml

```diff
@@ -3443,6 +3443,14 @@ Style/Dir:
   Enabled: true
   VersionAdded: '0.50'
 
+Style/DisableCopForEntireFileInSourceCode:
```

Comment:
I'm not sure about this name. Open to suggestions.

Reply:
`Lint/CopDisabledForEntireFile` is probably simpler IMO.

Reply:
Yes, I like that better. Thanks!

Reply:
`CopDisabledForEntireFile` is a good name 👍  But this seems to favor `Style` over `Lint`. `Lint` is the highest warning level, emulating Ruby warnings, preventing errors and deprecated APIs (e.g. that will be removed in the future) . For example, specifying `Exclude` in .rubocop.yml is practically equivalent but not as bad as `Lint` would detect it. It will depend on the usage preference of user (`Style`) .

Reply:
Honestly, there probably should be a separate `Rubocop`, etc. department for the cops that deal with rubocop directives, because `Style` is about code style to me, not about all preferences, and `Lint` is not currently limited to ruby warnings, plus has other directive cops as mentioned earlier.

I think it makes sense to be grouped with the related cops, but it doesn't really matter at the end of the day.

Reply:
Ah, I see. But the fact that `Style` and `Lint` have different warning levels is a point for users. This is a tough one...

Reply:
🤔 I see there is `InternalAffairs` for Cops that apply to Rubocop itself, although this is a cop that applies to Rubocop usage by consumers. Is there a name that would keep the "cop" theme? 😄 Maybe `Dispatch` or `Patrol` or something?

Either that, or we should decide on the criteria for if a cop is `Style` or `Lint` (if that isn't already documented already), and apply that to all the Rubocop cops.

Yet another option would be to follow the pattern of extracting a gem (`rubocop-rubocop`?), although this may be a special case as some of the cops catch actual mistakes (e.g. forgetting to re-enable a cop).